tests/kernel/mem_protect/userspace/prj.conf

```diff
@@ -1 +1,3 @@
 CONFIG_ZTEST=y
+CONFIG_INIT_STACKS=y
+CONFIG_ARC_STACK_CHECKING=n
```

Comment:
Ugh this is generating a fatal sanitycheck error, need to add a second test I guess

Reply:
Should we try and get stack_analyze() working for ARC, instead?
ARC hardware stack checking protects unallocated stack space and doesn't allow sp based accesses to it. Non sp based accesses should be fine...

Reply:
@ruuddw agreed, I'm looking into it

Reply:
A quick fix, by disabling stack bounds checking temporarily doesn't work - these control registers cannot be accessed in user mode. Checking the marker pattern through non stack-indexed registers should work, but might require some inline assembly.

Reply:
understood
stack_analyze() itself may need to change.

Reply:
fixed.